pwndbg/aglib/heap/ptmalloc.py

```diff
@@ -1598,36 +1624,41 @@ def thread_cache(self) -> pwndbg.dbg_mod.Value | None:
         """Locate a thread's tcache struct. If it doesn't have one, use the main
         thread's tcache.
         """
-        if self.has_tcache():
-            if self.multithreaded:
-                tcache_addr = pwndbg.aglib.memory.read_pointer_width(
-                    pwndbg.aglib.symbol.lookup_symbol_addr("tcache", prefer_static=True)
-                )
-                if tcache_addr == 0:
-                    # This thread doesn't have a tcache yet
-                    return None
-                tcache = tcache_addr
-            else:
-                tcache = self.main_arena.heaps[0].start + pwndbg.aglib.arch.ptrsize * 2
+        if not self.has_tcache():
+            print(message.warn("This version of GLIBC was not compiled with tcache support."))
+            return None
 
-            try:
-                self._thread_cache = pwndbg.aglib.memory.get_typed_pointer_value(
-                    self.tcache_perthread_struct, tcache
-                )
-                self._thread_cache["entries"].fetch_lazy()
-            except Exception:
-                print(
-                    message.error(
-                        "Error fetching tcache. GDB cannot access "
-                        "thread-local variables unless you compile with -lpthread."
-                    )
-                )
-                return None
+        tcache_ptr = pwndbg.aglib.symbol.lookup_symbol_addr(
+            "tcache",
+            objfile_endswith=pwndbg.glibc.get_libc_filename_from_info_sharedlibrary,
```

Comment:
Are we missing a call here?
```suggestion
            objfile_endswith=pwndbg.glibc.get_libc_filename_from_info_sharedlibrary(),
```

Reply:
yeah looks like it, should be fixed in latest commit